app/scripts/controllers/network/network-controller.ts

```diff
@@ -596,16 +596,6 @@ export class NetworkController extends EventEmitter {
       return;
     }
 
-    if (!chainId) {
```

Comment:
This scenario is now impossible. If a provider is set then `#configureProvider` must have been called and we would have already checked that `chainId` is set. (This check is not present on the core side.)